library/pkparse.c

```diff
@@ -1255,12 +1289,43 @@ int mbedtls_pk_parse_public_key( mbedtls_pk_context *ctx,
     mbedtls_pem_context pem;
 
     mbedtls_pem_init( &pem );
-
+#if defined(MBEDTLS_RSA_C)
     /* Avoid calling mbedtls_pem_read_buffer() on non-null-terminated string */
     if( keylen == 0 || key[keylen - 1] != '\0' )
         ret = MBEDTLS_ERR_PEM_NO_HEADER_FOOTER_PRESENT;
     else
         ret = mbedtls_pem_read_buffer( &pem,
+                               "-----BEGIN RSA PUBLIC KEY-----",
+                               "-----END RSA PUBLIC KEY-----",
+                               key, NULL, 0, &len );
+
+    if( ret == 0 )
+    {
+         const mbedtls_pk_info_t *pk_info;
+         if( ( pk_info = mbedtls_pk_info_from_type( MBEDTLS_PK_RSA ) ) == NULL )
+              return( MBEDTLS_ERR_PK_UNKNOWN_PK_ALG );
+
+         if( ( ret = mbedtls_pk_setup( ctx, pk_info ) ) != 0 )
+               return( ret );
+
+         if( ( ret = pk_parse_key_pkcs1_der( mbedtls_pk_rsa( *ctx ),
```

Comment:
You could call the existing `pk_get_rsapubkey` from here.

Reply:
I tried, `pk_get_rsapubkey` needs start and end, which are calculated in `pk_parse_key_pkcs1_der`

Reply:
@RonEld You can setup a local variable holding `pem.buf` initially and pass its address as `p`, and `pem.buf + pem.buflen` as `end`.

Reply:
@hanno-arm yes, I know, but I prefered not adding too much variables. If it's a blocker, I'll do it